### PR TITLE
Binary Search Tree

### PR DESCRIPTION
**Pull request by shravan-cc**
This pull request adds a JavaScript implementation of a Binary Search Tree (BST), consisting of a Tree class and a Node class. The Tree class offers methods for creating a tree, adding nodes, and performing traversals like in-order, pre-order, and post-order. It also provides a method for removing nodes from the tree while maintaining its structure.

### CODE REVIEW

File: tree.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
+/* eslint-disable import/prefer-default-export */
+
+class Node {
```
Comment (shripada) [inline tree.js:5]: I think this file is not pretty formatted, can you check? have you integrated prettier?

File: tree.js [hunk 2]
```diff
@@ -0,0 +1,191 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (shripada) [inline tree.js]: Avoid disabling eslint rules. Figure out why lint errors are coming and explore and make necessary changes.

File: tree.js [hunk 3]
```diff
@@ -0,0 +1,191 @@
+/* eslint-disable import/prefer-default-export */
+
+class Node {
+    constructor(data) {
+      this.data = data;
+      this.left = null;
+      this.right = null;
+    }
+  }
+  export class Tree {
```
Comment (shripada) [inline tree.js]: use jsdoc to document classes and methods

File: tree.js [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+      this.root = null;
+    }
+    
+    /**
+     * Initializes the binary search tree by setting the root node to null.
+     *
+     * @returns {null} - Always returns null.
+     */
+
+    createTree() {
```
Comment (shripada) [inline tree.js]: this is not needed, we should use constructor only, `new Tree()`

File: tree.js [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+    addItem(data) {
+      const newNode = new Node(data);
+      if (this.root === null) {
+        this.root = newNode;
+        return newNode;
+      }
+      let currentNode = this.root;
+      while (true) {
+        if (newNode.data < currentNode.data) {
```
Comment (shripada) [inline tree.js]: this comparison will work only with numbers/strings etc.  Ideally we need to pass a 'compare` function to the tree's constructor. That function should receive two data items to compare and it should return 0 if both equal, 1 if later is larger and -1 if former is larger.

**Review by shripada — CHANGES_REQUESTED** [review]

File: tree.js (hunk 1 above)
Comment (shravan-cc) [reply tree.js:5]: I made the neccassary changes.

File: tree.js (hunk 3 above)
Comment (shravan-cc) [reply tree.js]: Made the changes.

File: tree.js (hunk 4 above)
Comment (shravan-cc) [reply tree.js]: Implemented the changes.

File: tree.js (hunk 5 above)
Comment (shravan-cc) [reply tree.js]: Implemented the changes and modified by code.

File: tree.js (hunk 2 above)
Comment (shravan-cc) [reply tree.js]: Implemented the changes

**Pull request merged**